resources/default.rb

```diff
@@ -4,19 +4,59 @@
 property :artifactoryonline, :kind_of => String
 property :repository, :kind_of => String, :required => true
 property :repository_path, :kind_of => String, :required => true
+property :artifact_name, :kind_of => String, :required => true
 property :artifactory_username, :kind_of => String
 property :artifactory_password, :kind_of => String
+property :highest, :kind_of => [TrueClass, FalseClass]
 
 property :group, :kind_of => [Integer, String]
 property :mode, :kind_of => [Integer, String]
 property :owner, :kind_of => [Integer, String]
 
 action_class do
   require "uri"
+  require 'net/http'
+  require 'json'
 
   include ::ArtifactoryArtifact::Helper
 
+  def fetch_highest(new_resource)
+    # Fetching list of artifacts from artifactory
+    url = URI("#{new_resource.artifactory_url}/api/search/aql/")
+    http = Net::HTTP.new(url.host, url.port)
+    request = Net::HTTP::Post.new(url)
+
+    request['Content-Type'] = 'text/plain'
+    request['Accept'] = '*/*'
+    request['Host'] = url.host
+    request['Connection'] = 'keep-alive'
+    request.basic_auth new_resource.artifactory_username, new_resource.artifactory_password
+    request.body = "items.find({\"repo\":{\"$eq\":\"   #{new_resource.repository}\"}})"
+    response = http.request(request)
+
```

Comment:
As per my understanding, net/http won't raise exceptions even if remote server respond as 4xx or 5xx. It'd be better add some code here to verify return code.